### PR TITLE
1828: Allow the game class's cache to be cleared

### PR DESCRIPTION
**Pull request by crericha**
- Add function to clear cache of the specified type
- Use to clear share_prices cache when new par prices become unlocked.

### CODE REVIEW

File: lib/engine/game/base.rb [hunk 1]
```diff
@@ -1640,6 +1640,13 @@ def cache_objects
         end
       end
 
+      def clear_cache(type)
+        return unless CACHABLE.any? { |t, _n| t == type }
```
Comment (tobymao) [inline lib/engine/game/base.rb:1644]: i'm a bit wary of this function? what's the intention?

Comment (crericha) [reply lib/engine/game/base.rb:1644]: Short answer: to clear the cached par prices. 
Longer answer: In 1828, more par prices become available throughout the game and if the cache isn't updated with the new par prices, then share_price_by_id won't be able to get at them. The game crashes on the par action if the screen is refreshed or undo used.

Maybe update_cache is a better name that clear_cache?

Comment (crericha) [reply lib/engine/game/base.rb:1644]: Essentially, the cache implies all of those objects are immutable. That's not true for all games.

Comment (tobymao) [reply lib/engine/game/base.rb:1644]: ok change to update

Comment (crericha) [reply lib/engine/game/base.rb:1644]: Change made.

**Review by tobymao — APPROVED** [review]

**Pull request merged**